monitoring/voltage-monitoring/src/main/java/com/farao_community/farao/monitoring/voltage_monitoring/json/VoltageMonitoringResultDeserializer.java

```diff
@@ -92,4 +117,58 @@ private void readVoltageValues(JsonParser jsonParser, Map<VoltageCnec, ExtremeVo
             voltageValues.put(voltageCnec, new ExtremeVoltageValues(Set.of(min, max)));
         }
     }
+
+    private void readAppliedRas(JsonParser jsonParser, Map<State, Set<NetworkAction>> appliedRas) throws IOException {
+        while (jsonParser.nextToken() != JsonToken.END_ARRAY) {
+            String contingencyId = null;
+            Instant instant = null;
+            Set<String> remedialActionIds = new HashSet<>();
+            while (!jsonParser.nextToken().isStructEnd()) {
+                switch (jsonParser.currentName()) {
+                    case INSTANT:
+                        instant = deserializeInstant(jsonParser.nextTextValue());
+                        break;
+                    case CONTINGENCY:
+                        contingencyId = jsonParser.nextTextValue();
+                        break;
+                    case REMEDIAL_ACTIONS:
+                        jsonParser.nextToken();
+                        remedialActionIds = jsonParser.readValueAs(new TypeReference<HashSet<String>>() {
+                        });
+                        break;
+                    default:
+                        throw new FaraoException(String.format(UNEXPECTED_FIELD_ERROR, jsonParser.currentName(), REMEDIAL_ACTIONS));
+                }
+            }
+            if (instant == null) {
+                throw new FaraoException(String.format("Instant must be defined in %s", REMEDIAL_ACTIONS));
+            }
+            // Get network Actions from string
+            State state = getState(instant, contingencyId);
+            if (appliedRas.containsKey(state)) {
+                throw new FaraoException(String.format("State with instant %s and contingency %s has previously been defined in %s", instant.toString(), contingencyId, REMEDIAL_ACTIONS));
+            } else {
+                appliedRas.put(state, getNetworkActions(remedialActionIds));
+            }
+        }
+    }
+
+    private State getState(Instant instant, String contingencyId) {
+        if (Objects.isNull(contingencyId)) {
```

Comment:
Objects.isNull is only useful in lambdas, we won't be using it anymore in if statements
```suggestion
        if (contingencyId == null) {
```

Reply:
Done in the common code !